### ISSUE TITLE
Pa11y-ci support for WCAG 2.2

### ISSUE THREAD

**Issue by Shehreem-Solirius** [pa11y/pa11y-ci#205]
Will pa11y-ci be upgraded to support WCAG2.2 or if it already does, what is the tag for it?

**Comment by hollsk** [comment]
Hi Shehreem, thanks for using Pa11y! Yes, we will support WCAG 2.2 - we rely on the axe and HTML_Codesniffer rules engines for the test criteria, so the "when" depends partly on them, partly on us 😄 

HTML_CS doesn't support WCAG 2.2 yet, and although I don't know what their roadmap looks like, I wouldn't expect them to release a 2.2 ruleset before the guidelines have been finalised. 

The 4.5 version of axe core does support WCAG 2.2, but we're using an older version. The axe team have released their new rules much earlier than most - most devs are going to want to wait so they don't end up wasting time building rulesets that could be changed or removed. Deque just has more people to throw at this than the rest of us, so I imagine it's less of a problem for them 😅 


So, yes, hopefully we'll update that soon, and there's a chance that we might add support via axe before HTML_CS, which will depend on the Pa11y team's bandwidth.

**Comment by werdnanoslen** [comment]
Now that 2.2 has been [finalized](https://www.digitala11y.com/wcag-2-2-is-finally-here/), it [seems very unlikely](https://github.com/squizlabs/HTML_CodeSniffer/issues/315#issuecomment-1758703047) that HTML_CS will support WCAG 2.2, so I think you'll have to rely on axe. 

Any indication on when pa11y might support it then?

**Comment by josebolos** [comment]
Hi @werdnanoslen!

Not at the moment, no. We try to update pa11y to use the latest versions of HTML_CS and axe if and when we can, on a best-effort basis, as @hollsk indicated. I'm not aware of what level of WCAG 2.2 support axe offers at the moment.

We always try to make it as clear as possible what has changed whenever we release a new version, so I would suggest to keep an eye on the [Changelog](https://github.com/pa11y/pa11y/blob/main/CHANGELOG.md).

**Comment by werdnanoslen** [comment]
[axe has one rule](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#wcag-22-level-a--aa-rules) to check target size, but they do support it

**Comment by j-mendez** [comment]
> Will pa11y-ci be upgraded to support WCAG2.2 or if it already does, what is the tag for it?

For 2.2 and beyond [kayle](https://github.com/a11ywatch/kayle) handles this with a lot of bug fixes and speed improvements. It should be a drop in replacement. If you are using this on the cloud prepare for a huge cost savings since the lib has rewrite of axe and htmlcs that take performance to the next level.
